src/linux/processors.c

```diff
@@ -340,6 +368,38 @@ static bool siblings_parser(uint32_t sibling_list_start, uint32_t sibling_list_e
 	return context->callback(processor, sibling_list_start, sibling_list_end, context->callback_context);
 }
 
+bool cpuinfo_linux_detect_core_cpus(
+	uint32_t max_processors_count,
+	uint32_t processor,
+	cpuinfo_siblings_callback callback,
+	void* context)
+{
+	char core_cpus_filename[CORE_CPUS_FILENAME_SIZE];
+	const int chars_formatted = snprintf(
+		core_cpus_filename, CORE_CPUS_FILENAME_SIZE, CORE_CPUS_FILENAME_FORMAT, processor);
+	if ((unsigned int) chars_formatted >= CORE_CPUS_FILENAME_SIZE) {
+		cpuinfo_log_warning("failed to format filename for core cpus of processor %"PRIu32, processor);
+		return false;
+	}
+
+	struct siblings_context siblings_context = {
+		.group_name = "cpus",
+		.max_processors_count = max_processors_count,
+		.processor = processor,
+		.callback = callback,
+		.callback_context = context,
+	};
+	if (cpuinfo_linux_parse_cpulist(core_cpus_filename,
+		(cpuinfo_cpulist_callback) siblings_parser, &siblings_context))
+	{
```

Comment:
Keep opening curly bracket on the same line as if statement
```suggestion
		(cpuinfo_cpulist_callback) siblings_parser, &siblings_context)) {
```

Reply:
Fixed, here and elsewhere.